[ResizeObserver] The initial value of the last reported size should be -1x-1

cathiechen opened the pull request:
#### 91f24ff5be182c5595eec41e8e4718c94ae0db64
<pre>
[ResizeObserver] The initial value of the last reported size should be -1x-1
<a href="https://bugs.webkit.org/show_bug.cgi?id=250836">https://bugs.webkit.org/show_bug.cgi?id=250836</a>

Reviewed by NOBODY (OOPS!).

The conclusion of [1] indicates that when first observing an element with ResizeObserver,
lastReportedSize gets initialized with a -1 x -1 size.

[1] <a href="https://github.com/w3c/csswg-drafts/issues/3664">https://github.com/w3c/csswg-drafts/issues/3664</a>

* LayoutTests/imported/w3c/web-platform-tests/resize-observer/notify-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/resize-observer/observe-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/resize-observer/svg-expected.txt:
* LayoutTests/platform/mac-wk1/imported/w3c/web-platform-tests/resize-observer/notify-expected.txt: Copied from LayoutTests/imported/w3c/web-platform-tests/resize-observer/notify-expected.txt.
* LayoutTests/platform/mac/imported/w3c/web-platform-tests/resize-observer/svg-expected.txt:
* Source/WebCore/page/ResizeObservation.cpp:
</pre><!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/91f24ff5be182c5595eec41e8e4718c94ae0db64

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [❌ 🧪 style](https://ews-build.webkit.org/#/builders/6/builds/104378 "1 style error") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/77/builds/13459 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/43/builds/37289 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/8/builds/113595 "Built successfully") | [✅ 🛠 🧪 win](https://ews-build.webkit.org/#/builders/10/builds/173888 "Built successfully and passed tests") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/11/builds/108302 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/76/builds/14559 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/85/builds/4374 "Built successfully") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/36/builds/96602 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/12/builds/112635 "Built successfully") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/19/builds/110145 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/78/builds/11223 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/3/builds/94284 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/35/builds/38853 "Passed tests") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/9/builds/93086 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/73/builds/25895 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/34/builds/80522 "Passed tests") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/81/builds/6819 "Built successfully") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/70/builds/27252 "Passed tests") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/82/builds/6949 "Built successfully") | [❌ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/84/builds/3802 "Found 2 new test failures: fast/images/avif-heif-container-as-image.html, resize-observer/observe-disconnected-target-crash.html (failure)") | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/80/builds/12974 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/62/builds/46810 "Passed tests") | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/79/builds/8742 "Built successfully") | | | 
| | | | | 
<!--EWS-Status-Bubble-End-->

Loirooriol reviewed (approved):
LGTM if you fix the commit message

Loirooriol reviewed (commented):
You edited the comment, but the commit was not updated. I suspect this won't do.

Loirooriol reviewed (changes requested):
I guess [observe-disconnected-target-crash.html](https://github.com/WebKit/WebKit/blob/main/LayoutTests/resize-observer/observe-disconnected-target-crash.html) fails because you need something like https://github.com/WebKit/Webkit/commit/1a04d71fd0c83f072d19295920c6c9c1ac8cfadd

cathiechen committed:
[ResizeObserver] The initial value of the last reported size should be -1x-1
https://bugs.webkit.org/show_bug.cgi?id=250836

Reviewed by NOBODY (OOPS!).

The conclusion of [1] indicates that when first observing an element with ResizeObserver,
lastReportedSize gets initialized with a -1 x -1 size.

[1] https://github.com/w3c/csswg-drafts/issues/3664

* LayoutTests/imported/w3c/web-platform-tests/resize-observer/notify-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/resize-observer/observe-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/resize-observer/svg-expected.txt:
* LayoutTests/platform/mac-wk1/imported/w3c/web-platform-tests/resize-observer/notify-expected.txt: Copied from LayoutTests/imported/w3c/web-platform-tests/resize-observer/notify-expected.txt.
* LayoutTests/platform/mac/imported/w3c/web-platform-tests/resize-observer/svg-expected.txt:
* Source/WebCore/page/ResizeObservation.cpp:
